DashAI/back/api/api_v1/endpoints/runner.py

```diff
@@ -51,51 +53,35 @@ async def execute_run(run_id: int, db: Session = Depends(get_db)):
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
             detail="Internal database error",
         )
-    # Instance the model with the run parameters
+    # Dataset
+    dataset = load_dataset(f"{dataset.file_path}/dataset")
+
+    # Model
     run_model_class = component_registry[run.model_name]["class"]
-    run_instance = run_model_class(**run.parameters)
-    dataset_dict = load_dataset(f"{dataset.file_path}/dataset")
-    # Prepare dataset
-    dataset_dict = component_registry[exp.task_name]["class"]().prepare_for_task(
-        dataset_dict
-    )
-    # Format dataset: TBD move this to dataloader
-    dataset_dict = run_instance.format_data(dataset_dict)
-    # Training
-    run.run_start()
-    db.commit()
-    run_instance.fit(dataset_dict["train"]["input"], dataset_dict["train"]["output"])
-    # Evaluation
+    model = run_model_class(**run.parameters)
+
+    # Task
+    task = component_registry[exp.task_name]["class"]()
+
+    # Get evaluation Metrics
     metrics = _intersect_component_lists(
         component_registry.get_components_by_types(select="Metric"),
         component_registry.get_related_components(exp.task_name),
-    )
-    run.train_metrics: Dict[str, DashAIDataset] = {
-        metric_name: metric_cls.score(
-            dataset_dict["train"]["output"],
-            run_instance.predict(dataset_dict["train"]["input"]),
-        )
-        for metric_name, metric_cls in metrics.items()
-    }
-    run.validation_metrics: Dict[str, DashAIDataset] = {
-        metric_name: metric_cls.score(
-            dataset_dict["validation"]["output"],
-            run_instance.predict(dataset_dict["validation"]["input"]),
-        )
-        for metric_name, metric_cls in metrics.items()
-    }
-    run.test_metrics = {
-        metric_name: metric_cls.score(
-            dataset_dict["test"]["output"],
-            run_instance.predict(dataset_dict["test"]["input"]),
-        )
-        for metric_name, metric_cls in metrics.items()
-    }
-    run.run_end()
-    # Save changes
-    run_path = f"{settings.USER_RUN_PATH}/{run.id}"
-    run_instance.save(run_path)
-    run.run_path = run_path
+    ).values()
+
+    # Mark run as delivered
+    run.run_delivery()
     db.commit()
 
+    # Execute the run
+    background_tasks.add_task(
```

Comment:
What happens if an exception is raised inside the background task?
Where do you set a run as failed?
My guess is that it is missing a try-except somewhere. 
This try-except should log some error message and modify the run status as failed.